sympy/solvers/solveset.py

```diff
@@ -854,6 +862,9 @@ def solveset(f, symbol=None):
 
     f = sympify(f)
 
+    if f is S.false:
```

Comment:
That is what i said.
if i use `if not f`
```
>>> solveset(0, x)
EmptySet()               # this is returned
```

if i use `if f is S.false`
```
>>> solveset(0, x)
(-oo, oo)
```
Hope this clarifies ?